### DISCUSSION TITLE
Please release a new version

### DISCUSSION THREAD

**Discussion by sylvainfaivre** [geany/geany#3499] [General]
Geany 1.38 was released in October 2021.
Lots of changes happened since, and people would be interested in them.

**Comment by quiltedstars** [comment]
Agree, especially the tree list view for documents!!

**Comment by satmandu** [comment]
At Chromebrew I just went ahead and built a new release from git:

https://github.com/chromebrew/chromebrew/blob/master/packages/geany.rb
https://github.com/chromebrew/chromebrew/blob/master/packages/geany_plugins.rb

**Comment by elextr** [comment]
First of all, note that there is nothing stopping distros from building from Git, several already do.

@b4n and @eht16 are the ones on which the work falls, its up to them when they have the available time (my timezone allows me to sleep through it :-).

**Comment by satmandu** [comment]
Thanks for all your work!

**Comment by TiZ-HugLife** [comment]
Hi, I'm one of the maintainers for Geany's Flathub package. Flathub generally really dislikes shipping regular git builds that aren't releases, mainly because at that point, everything becomes very arbitrary and difficult for maintainers. How do we determine which commit to build from? How often are we expected to pick a "canonical" git commit for the Flathub package? Weekly? Bi-weekly? Monthly? Bi-monthly? Are we expected to support users who pick their own arbitrary commits to build from?

Exceptions can be made, but it's much easier if there is some sort of anchor or waypoint. For example, the Dolphin emulator also has not made a release in over six years, coming up soon on seven. It's safe to assume they will likely just never make a release ever again. But they do monthly progress reports; that is the waypoint around which builds from git are done. The latest commit at the time of publishing a progress report is used to ship git builds of Dolphin. This sounds risky--how can you be sure that the software is in a working state during that commit?--but I believe that Dolphin's developers have a policy of making sure that git master is always usable, doing their heavy lifting, disruptive work in PRs. I don't know if you guys do that too. It's honestly not really my job to know.

So if there's anything you all can do to make choosing git commits more tenable for package maintainers, that will make it feasible for us to ship it in Flathub. I understand that you all probably don't want to be doing progress report blog posts, but if you could make some periodic tags for known good commits, that would probably be good enough for us to start shipping git without having to hold any conversations about it.

**Comment by elextr** [comment]
One of the complications for Geany is its available on three platforms, a couple of billion[^1] Linux distros and other distribution mechanisms (flatsnapapps etc), Windows, and Macos.  

The latter two do not have a culture of "build your own" nor suitably open distribution mechanisms for people to build and distribute third party software.  There is also no significant contribution to those platforms[^2] so the Geany project makes the packages itself[^3].  Well, actually each platform is one guy, the Macos guy uses Geany on Macos AFAIK, but the Windows guy doesn't even use Geany on Windows himself, so the build is an unusual mix of cross building, docker containers, and fairy dust ... and broken.

At the moment the Windows build process is undergoing an upgrade that should make it easier for others to support and for Windows builds to be made by CI so people can test PRs on Windows.  AFAIK the Macos process is currently working, albeit manually.

So releases have necessarily been delayed until all three platforms can be released at the same time, there is not enough effort available to support different versions of Geany on different platforms, and tagging just for Linux would compound that problem since it will require tracking exactly what features/bug/fixes were in the tag, assuming the person raising the issue actually provided the tag.  So supporting a number of stable tags isn't an acceptable process.

All Geany changes are done by PR, and the main Geany repository is deliberately kept buildable.  CI ensures that, and that Geany at least starts before any PR is merged.  But extensively testing an aggressively GUI application on CI isn't possible.  Nonetheless git Geany mostly works fine, most of the developers use it.   

AFAIK so far the distros that build from git are mostly rolling releases, which are used to things stopping working every second Thursday and getting fixed on Tuesday week, but Geany should be much more stable than that.

All of which does not help the more stable channels I know, but at least you have the background.

[^1]: slight exaggeration for effect.
[^2]: in fact a previous Windows contributor has just reappeared as a Linux contributor, thats one less for Windows.
[^3]: being a grumpy olde guy I say drop those platforms unless a user steps up to support them properly, but nicer people prevailed :-)

**Reply by satmandu** [reply]
Sounds like the macos build process needs updating then? It is possible to automate on GitHub. I think jellyfin-media-player has builds done by GitHub for macos. (The last two versions of macos are supported by the GitHub builders.)

Maybe after the windows build is redone it might make sense to tag a version? (Also maybe that's a good time to target getting macos builds done on GitHub?)

**Comment by asoymos** [comment]
Thank you for explanation.

**Comment by eht16** [comment]
Here we go, Geany 2.0 waits for you to be installed or upgraded:
https://www.geany.org/news/geany-20-is-out/

It's bad it took so long but hopefully it will be the best release ever and next one will be faster :).
